ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java

```diff
@@ -40,25 +57,143 @@ private ListQueriesExecutor() { }
   ) {
     final ListQueries listQueries = statement.getStatement();
     if (listQueries.getShowExtended()) {
-      return Optional.of(new QueryDescriptionList(
-          statement.getStatementText(),
-          executionContext.getPersistentQueries().stream()
-              .map(QueryDescriptionFactory::forQueryMetadata)
-              .collect(Collectors.toList())));
+      return executeExtended(
+          listQueries,
+          statement,
+          sessionProperties,
+          executionContext,
+          serviceContext);
+    }
+
+    final List<RunningQuery> runningQueries = executionContext.getPersistentQueries()
+        .stream()
+        .map(q -> {
+          final KafkaStreamsStateCount kafkaStreamsStateCount = new KafkaStreamsStateCount();
+          kafkaStreamsStateCount.updateStateCount(q.getState(), 1);
+          return new RunningQuery(
+              q.getStatementString(),
+              ImmutableSet.of(q.getSinkName().text()),
+              ImmutableSet.of(q.getResultTopic().getKafkaTopicName()),
+              q.getQueryId(),
+              kafkaStreamsStateCount
+          );
+        }).collect(Collectors.toList());
+
+    final Map<String, RunningQuery> queryToRunningQuery = runningQueries.stream().collect(
+        Collectors.toMap(
+            query -> query.getId().toString(),
+            query -> query));
+
+    if (!sessionProperties.getInternalRequest()) {
+      final Set<HostInfo> hosts =
+          DiscoverRemoteHostsUtil.getRemoteHosts(
+              executionContext.getPersistentQueries(),
+              sessionProperties.getKsqlHostInfo());
+
+      hosts.forEach(hostInfo -> {
+        final KsqlEntityList response = serviceContext.getKsqlClient()
+            .makeKsqlRequestWithRequestProperties(
+                ServerUtil.buildRemoteUri(
+                    sessionProperties.getLocalUrl(),
+                    hostInfo.host(),
+                    hostInfo.port()
+                ),
+                statement.getStatementText(),
+                Collections.singletonMap(KsqlRequestConfig.KSQL_REQUEST_INTERNAL_REQUEST, true))
+            .getResponse();
+
+        response.forEach(remoteQueries -> {
+          final List<RunningQuery> queries = ((Queries) remoteQueries).getQueries();
+          queries.forEach(q -> {
+            final String queryId = q.getId().toString();
+            
+            // If the query has already been discovered, update the KafkaStreamsStateCount object
+            if (queryToRunningQuery.containsKey(queryId)) {
+              q.getState().getState()
+                  .forEach((state, count) ->
```

Comment:
nit: up to you, but I would recommend replacing all of these `.forEach` calls with vanilla for-each loops. It will make the stack traces way more readable, error handling easier and debugging anything with remote calls a little simpler. the way it stands, I think we have four nested lambdas, which were really designed for stateless transforms

Reply:
done